common/persistence/cassandra/nexus_incoming_service_store.go

```diff
@@ -230,7 +230,7 @@ func (s *NexusIncomingServiceStore) ListNexusIncomingServices(
 
 	response.TableVersion = currentTableVersion
 
-	if request.LastKnownTableVersion != currentTableVersion {
+	if request.LastKnownTableVersion != 0 && request.LastKnownTableVersion != currentTableVersion {
```

Comment:
I'd add a comment explaining when this would be 0 and why we added this condition.